### PR TITLE
Support for tows c block

### PR DESCRIPTION
**Pull request by diptighosalkar**

### CODE REVIEW

File: webapp/data_structures/TOWS_c.js [hunk 1]
```diff
@@ -41,33 +41,140 @@ function TOWS_c() {
...
+            herit2 = 1;
+        }
+        nz2 = parseFloat(nz2);
+        this.nz = nz1;
+        this.varnam = varnam1;
+        this.herit = herit1;
+        this.x.model.intyp = new ScilabDouble([-1]);
+        var io = set_io(this.x.model,this.x.graphics,[[-1],[-2]],[],ones(1-herit2,1),[]);
+        if(herit2 == 1){
+            this.x.model.blocktype = new ScilabDouble(["x"]);
```
Comment (sunilshetye) [inline webapp/data_structures/TOWS_c.js]:
```suggestion
            this.x.model.blocktype = new ScilabString(["x"]);
```

blocktype should be string, not double

File: webapp/data_structures/TOWS_c.js [hunk 2]
```diff
@@ -41,33 +41,140 @@ function TOWS_c() {
...
+        }else{
             this.x.model.blocktype = new ScilabDouble(["d"]);
```
Comment (sunilshetye) [inline webapp/data_structures/TOWS_c.js]:
```suggestion
            this.x.model.blocktype = new ScilabString(["d"]);
```

blocktype should be string, not double

File: webapp/data_structures/TOWS_c.js [hunk 3]
```diff
@@ -41,33 +41,140 @@ function TOWS_c() {
...
+                throw "incorrect";
+            }
+        }else{
+            alert("Invalid variable name. \nPlease choose another variable name.");
+            throw "incorrect";
+        }
+
+        herit1 = herit1.trim().replace(regex_parentheses, '');
+        if (!regex_num.test(herit1)) {
+            var chararray = herit1.match(regex_char);
```
Comment (sunilshetye) [inline webapp/data_structures/TOWS_c.js]: herit has value of 0 or 1 only. This is not correct.

File: webapp/data_structures/TOWS_c.js [hunk 4]
```diff
@@ -41,33 +41,140 @@ function TOWS_c() {
...
+                        throw "incorrect";
+                    }else{
+                        alert("Answer given for Inherit (no:0, yes:1)\nis incorrect: Missing operator, comma, or semicolon.");
+                        throw "incorrect";
+                    }
+                }
+            }
+            var semi_comma_array = herit1.match(regex_semicolon_comma);
+            if (semi_comma_array != null) {
+                alert("Answer given for Inherit (no:0, yes:1)\nhas invalid dimension:\nwaiting for dimension 1.");
```
Comment (sunilshetye) [inline webapp/data_structures/TOWS_c.js]: herit is a simple value of 0/1. It should not have a dimension.

File: webapp/data_structures/TOWS_c.js [hunk 5]
```diff
@@ -41,33 +41,140 @@ function TOWS_c() {
...
+                throw "incorrect";
+            }
         }
-        this.x.model.intyp = new ScilabDouble([-1])
-        var io = set_io(this.x.model,this.x.graphics,[[-1],[-2]],[],ones(1-this.herit,1),[])
-        if(this.herit == 1){
-            this.x.model.blocktype = new ScilabDouble(["x"]);
+        var herit2 = parseFloat(getValueOfImaginaryInput(herit1));
```
Comment (sunilshetye) [inline webapp/data_structures/TOWS_c.js]: herit should not be imaginary.

File: webapp/data_structures/TOWS_c.js (hunk 3 above)
Comment (diptighosalkar) [reply webapp/data_structures/TOWS_c.js]: It actually allows number like 23 or any such number as well and on char value it gives error. Thus this was implemented for this.

File: webapp/data_structures/TOWS_c.js (hunk 4 above)
Comment (diptighosalkar) [reply webapp/data_structures/TOWS_c.js]: this is for validation purpose

File: webapp/data_structures/TOWS_c.js (hunk 5 above)
Comment (diptighosalkar) [reply webapp/data_structures/TOWS_c.js]: this not only handle imaginary parts it also handles %pi and any other mathematical expression like 2*3 , inherit does allow all this(simple math exp)expression also

**Pull request merged**